### ISSUE TITLE
 jupyter-gallery with local jupyterhub

### ISSUE THREAD

**Issue by whisere** [gedankenstuecke/jupyter-gallery#62]
Hello I am trying to set up jupyter-gallery with a local jupyterhub instance.

Have followed the instruction here: https://github.com/OpenHumans/jupyter-gallery

and tryied both 
JUPYTERHUB_BASE_URL = http://localhost:8888
and JUPYTERHUB_BASE_URL = real IP:port

But the notebook list under gallery is still empty, and the share option is still not showing in jupyterhub notebook or lab. Anything else needs to be setup before jupyter-gallery can work with a local jupyterhub instance? Thanks.

**Comment by gedankenstuecke** [comment]
Hey @whisere,
Did you follow all the setup instructions to enable the sharing in your local JupyterHub instance as well, that is enabling the 3 server extensions? In that case you should at least see the share function in your individual notebooks I think!

**Comment by whisere** [comment]
Thanks @gedankenstuecke I did run all the Jupyter commands successfully from the instruction here: https://github.com/OpenHumans/jupyter-gallery

jupyter bundlerextension enable --py oh_bundler
jupyter nbextension install --py oh_bundler --sys-prefix
jupyter nbextension enable --py oh_bundler --sys-prefix
jupyter serverextension enable --py oh_notebook_importer

Any others missing? I use the Jupyterhub docker with NativeAuthenticator. Thanks!

**Comment by gedankenstuecke** [comment]
Did you also set the environmental variables for the connection to Open Humans? The notebooks for the Gallery are currently stored in Open Humans when being exported from the hub to act as storage backend before the public sharing. In our deployment that was easy to do as the authenticator is an OAuth authenticator that uses OH for the login (see https://github.com/wildtreetech/ohjh for our setup of the JH itself). 

I fear there's some extra work needed to make it work with a different setup at the moment.

**Comment by whisere** [comment]
Yes I did set these variables in the .env file:
OH_ACTIVITY_PAGE='https://www.openhumans.org/activity/your-project-name-should-be-here/'
OH_CLIENT_ID='id'
OH_CLIENT_SECRET='secret'

When I click login on the gallery, it goes to OpenHuman and authorized successfully, but when coming back to the gallery front page, the login button still shows login instead of logout, and nothing changes. 

And does it only work with singleuser jupyterhub image? Is it possible to work with NativeAuthenticator? Thanks.

**Comment by gedankenstuecke** [comment]
Yeah, I don't think it currently works with the NativeAuthenticator, but I'd be happy to consider pull requests if you want to add that functionality :-)

**Comment by whisere** [comment]
No problem, will check it out, any suggestion on how to implement it? And any idea on the login issue above? Thanks.

**Comment by gedankenstuecke** [comment]
For the implementation: Unfortunately not, the main thing will be to find out where to store the notebooks when they are exported, as you need some place to send them to. Right now the Gallery doesn't have it's own storage space for those.

For the `.env` file: Did you go to https://www.openhumans.org and register an OAuth2 application for your version of the gallery? You need to get some actual CLIENT_ID & CLIENT_SECRET etc. to make the login work.

**Comment by whisere** [comment]
Okay thanks, I will also look at https://github.com/wildtreetech/ohjh.

Yes I did register the OAuth2 at openhumans, and and set the client id and secret, it was authorized successfully at openhumans after clicking login at gallery, but nothing changes (login still shows login not logout) after redirected back to the gallery front page from openhumans.
